bridge/client/app/_views/ktb-environment-view/ktb-stage-overview/ktb-stage-overview.component.ts

```diff
@@ -1,74 +1,49 @@
-import { distinctUntilChanged, filter, map, switchMap, takeUntil, withLatestFrom } from 'rxjs/operators';
-import { ActivatedRoute, Router } from '@angular/router';
-import { combineLatest, Subject } from 'rxjs';
 import { DtFilterFieldDefaultDataSourceAutocomplete } from '@dynatrace/barista-components/filter-field/src/filter-field-default-data-source';
 import { ServiceFilterType } from '../ktb-stage-details/ktb-stage-details.component';
+import { ISelectedStageInfo } from '../ktb-environment-view.component';
 
 @Component({
   selector: 'ktb-stage-overview',
   templateUrl: './ktb-stage-overview.component.html',
   styleUrls: ['./ktb-stage-overview.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class KtbStageOverviewComponent implements AfterContentInit, OnDestroy {
+export class KtbStageOverviewComponent {
   public _dataSource = new DtFilterFieldDefaultDataSource();
   public filter: DtFilterArray[] = [];
   public isTriggerSequenceOpen: boolean;
   private filteredServices: string[] = [];
   private globalFilter: { [projectName: string]: { services: string[] } } = {};
-  private unsubscribe$: Subject<void> = new Subject<void>();
+  private _project?: Project;
 
-  public project$ = this.route.params.pipe(
-    map((params) => params.projectName),
-    filter((projectName): projectName is string => !!projectName),
-    distinctUntilChanged(),
-    switchMap((projectName) => this.dataService.getProject(projectName)),
-    filter((project): project is Project => !!project)
-  );
+  @Input() selectedStageInfo?: ISelectedStageInfo;
 
-  public readonly selectedStageName$ = this.route.paramMap.pipe(
-    map((params) => params.get('stageName')),
-    withLatestFrom(this.project$),
-    filter(([stageName, project]) => Boolean(stageName && project)),
-    map(([stageName]) => stageName)
-  );
-
-  private readonly paramFilterType$ = this.route.queryParamMap.pipe(map((params) => params.get('filterType')));
+  @Input() set project(project: Project | undefined) {
+    if (project) {
+      this.setFilter(project, this._project?.projectName !== project.projectName);
+    }
+    this._project = project;
+  }
+  get project(): Project | undefined {
+    return this._project;
+  }
 
-  @Output() selectedStageChange: EventEmitter<{ stage: Stage; filterType: ServiceFilterType }> = new EventEmitter();
+  @Output() selectedStageInfoChange: EventEmitter<ISelectedStageInfo> = new EventEmitter();
   @Output() filteredServicesChange: EventEmitter<string[]> = new EventEmitter<string[]>();
 
-  constructor(
-    private dataService: DataService,
-    private apiService: ApiService,
-    private route: ActivatedRoute,
-    private router: Router
-  ) {
+  constructor(private dataService: DataService, private apiService: ApiService) {
     this.isTriggerSequenceOpen = this.dataService.isTriggerSequenceOpen;
     this.dataService.isTriggerSequenceOpen = false;
```

Comment:
We should refactor this `isTriggerSequenceOpen` into a behavior subject and also provide it as an input parameter. Then we can also get rid of the DataService.